base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable = Nameable.new)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end

```

Comment:
- [ ] Please make sure the `nameable` variable is readable and writable so it's accessible for read and write.
  - You can consider using `attr_accessor :nameable`